packages/grafana-ui/src/components/DataSourceSettings/DataSourceHttpSettings.tsx

```diff
@@ -207,6 +207,22 @@ export const DataSourceHttpSettings: React.FC<HttpSettingsProps> = (props) => {
             />
           </div>
 
+          {azureSettings?.azureAuthEnabled && (
```

Comment:
I think instead of adding a  "special case" property like this we should probably refactor the `DataSourceHttpSettings` and break out smaller reusable components that can be reused in a new `AzureDataSourceHttpSettings` and `DataSourceHttpSettings`. 

We are happy to help out with this if it feels like a little to big task to do within this PR.

Reply:
What is your thoughts on this @jackw?

Reply:
Like we discussed after this, we're going to stay with the general approach in this PR for now, acknowledging that this is a middleground for where we would like to get to (in a few aspects).

@grafana/plugins-platform - it would be great to get a path forward for how any data source can extend `DataSourceHttpSettings` to add additional configuration, such as SigV4 or Azure authentication.